include/string.h

```diff
@@ -162,11 +178,207 @@ static inline int strncmp(const char *s1, const char *s2, size_t n) {
+
+    if (NULL == s2)
+        return NULL;
+
+    strcpy(s2, s1);
+    return s2;
+}
+
+static inline char *strchr(const char *s, int c) {
+    if (NULL == s)
+        return NULL;
+
+    while (*s != (char) c) {
+        if ('\0' == *s)
+            return NULL;
+        s++;
+    }
+
+    return (char *) s;
+}
+
+static inline size_t strspn(const char *s1, const char *s2) {
+    size_t ret = 0;
+
+    if (NULL == s1 || NULL == s2)
+        return 0;
+
+    while (*s1 && strchr(s2, *s1)) {
+        s1++;
+        ret++;
+    }
+
+    return ret;
+}
+
+static inline size_t strcspn(const char *s1, const char *s2) {
+    size_t ret = 0;
+
+    if (NULL == s1 || NULL == s2)
+        return 0;
+
+    while (*s1) {
+        if (strchr(s2, *s1))
+            return ret;
+        else {
+            s1++;
+            ret++;
+        }
+    }
+
+    return ret;
+}
+
+static inline char *strtok(char *s, char *delim) {
+
+    static char *lasts;
+    int ch;
+
+    if (s == 0)
+        s = lasts;
+
+    do {
+        if ((ch = *s++) == '\0')
+            return 0;
+    } while (strchr(delim, ch));
+
+    s--;
+    lasts = s + strcspn(s, delim);
+
+    if (*lasts != 0)
+        *lasts = 0;
+
+    return s;
+}
+
+static inline char *strpbrk(const char *s, const char *chars) {
+    char c;
+    const char *p;
+
+    for (c = *s; c != 0; s++, c = *s) {
+        for (p = chars; *p != 0; p++) {
+            if (c == *p) {
+                return (char *) s;
+            }
+        }
+    }
+    return NULL;
+}
+
+static inline char *strstr(const char *s1, const char *s2) {
+    char *a = NULL, *b = NULL;
+
+    b = (char *) s2;
+    /* s2 is NULL terminated, return entire s1 */
+    if ('\0' == *b) {
+        return (char *) s1;
+    }
+
+    /* first char match in s1 */
+    for (; *s1 != '\0'; s1++) {
+        if (*s1 != *b) {
+            continue;
+        }
+        /* record the position */
+        a = (char *) s1;
+
+        while (1) {
+            /* s2 has been searched, return s1 */
+            if (*b == '\0') {
+                return (char *) s1;
+            }
+            /* no match, break and find new a */
+            if (*a++ != *b++) {
+                break;
+            }
+        }
+        /* rewind b all the way back to s2 start */
+        b = (char *) s2;
+    }
+
+    return NULL;
+}
+
+static inline char *strsep(char **str, char *delim) {
+    char *spanp = NULL, *s = NULL;
+    int c, sc;
+    char *tok = NULL;
+
+    s = *str;
+    if (NULL == s)
+        return NULL;
+
+    for (tok = s;; ++s) {
+        c = *s;
+        spanp = delim;
+        do {
+            if ((sc = *spanp++) == c) {
+                if (c == 0) {
+                    *str = NULL;
+                    return (tok == s ? NULL : tok);
+                }
+                *s++ = '\0';
+                *str = s;
+                return (tok);
+            }
+        } while (sc);
+    }
+}
+
+static inline int strcasecmp(const char *s1, const char *s2) {
```

Comment:
this function could be a special case wrapper around `strncasecmp()`

Reply:
But that means I'll have to loop around both strings, find highest 'n' and then call strncasecmp